### ISSUE TITLE
StackOverflowError in PlayReload on 'run' command (Windows)

### ISSUE THREAD

**Issue by anikolaev** [playframework/playframework#4146]
I am not the Play/Scala developer but I need to run Play project. After some update I start getting the following error on 'sbt run':

```
2015-03-27 22:44:28,108 - [INFO] - from play in pool-4-thread-1 
Listening for HTTP on /0:0:0:0:0:0:0:0:9000

2015-03-27 22:44:41,146 - [ERROR] - from application in New I/O worker #1 


! Internal server error, for (GET) [/report/organizationList/13a719ae-6570-419f-bb6f-4c9ca2bcc16e/reportApp_instance1?startTime=1426464000001&endTime=1427061599999] ->

scala.MatchError: java.lang.StackOverflowError (of class java.lang.StackOverflowError)
    at play.PlayReload$$anonfun$taskFailureHandler$1.apply(PlayReload.scala:44) ~[na:na]
    at play.PlayReload$$anonfun$taskFailureHandler$1.apply(PlayReload.scala:44) ~[na:na]
    at scala.Option.map(Option.scala:145) ~[scala-library-2.11.6.jar:na]
    at play.PlayReload$.taskFailureHandler(PlayReload.scala:44) ~[na:na]
    at play.PlayReload$.compileFailure(PlayReload.scala:40) ~[na:na]
    at play.PlayReload$$anonfun$compile$1.apply(PlayReload.scala:17) ~[na:na]
    at play.PlayReload$$anonfun$compile$1.apply(PlayReload.scala:17) ~[na:na]
    at scala.util.Either$LeftProjection.map(Either.scala:377) ~[scala-library-2.11.6.jar:na]
    at play.PlayReload$.compile(PlayReload.scala:17) ~[na:na]
    at play.PlayRun$$anonfun$playRunTask$1$$anonfun$apply$2$$anonfun$apply$3$$anonfun$2.apply(PlayRun.scala:61) ~[na:na]
    at play.PlayRun$$anonfun$playRunTask$1$$anonfun$apply$2$$anonfun$apply$3$$anonfun$2.apply(PlayRun.scala:61) ~[na:na]
    at play.runsupport.Reloader.reload(Reloader.scala:295) ~[na:na]
    at play.core.ReloadableApplication$$anonfun$get$1.apply(ApplicationProvider.scala:122) ~[play_2.11-2.3.8.jar:2.3.8]
    at play.core.ReloadableApplication$$anonfun$get$1.apply(ApplicationProvider.scala:120) ~[play_2.11-2.3.8.jar:2.3.8]
    at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24) ~[scala-library-2.11.6.jar:na]
    at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24) ~[scala-library-2.11.6.jar:na]
    at scala.concurrent.forkjoin.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1361) ~[scala-library-2.11.6.jar:na]
    at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260) ~[scala-library-2.11.6.jar:na]
    at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339) ~[scala-library-2.11.6.jar:na]
    at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979) ~[scala-library-2.11.6.jar:na]
    at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107) ~[scala-library-2.11.6.jar:na]

2015-03-27 22:44:41,218 - [WARN] - from play in New I/O worker #1 
No application found at invoker init
```
There is also long log of console output here: https://gist.github.com/anikolaev/724a50736a89d30d2f0d

I have:
Windows 7, 
SBT 0.13.5, 
Scala 2.11.6 (tried with 2.11.2 too), 
SBT Play plugin: 2.3.8 (tried with 2.3.2 too). 
IDEA 13 (but I also tried to run 'sbt run' from console).

My colleagues with Ubuntu don't have such problem so it is something wrong with Windows or with my computer configuration.

I tried to find the place in source code which triggers this error. There are really weird things. I cannot provide the code here but there are lines like 
```
        orgIds <- Try (orgList.map(_.orgId))
```
and when I simply add another line
```
    orgIds <- Try (orgList.map(_.orgId))
```
or even 
```
    a <- Try (Nil)
```
error above appears.

Please help. I really need to either fix it or find some workaround. As this is probably related to Play reloader it it possible to switch off reloading at all? It is ok for me to run this project once and don't touch it or rerun if required.

**Comment by shkhln** [comment]
This is actually a compiler bug (and the place for those is https://issues.scala-lang.org). However I see [another user with a similar problem](https://groups.google.com/forum/#!topic/play-framework/XZ-1F5bKdHg) on the mailing list, which is probably worth mentioning here just in case.

**Comment by shkhln** [comment]
> This is actually a compiler bug

On the second glance, I don't see activator startup script specifying thread stack size anywhere, and the default stack size on 32-bit windows jvm is quite a bit [smaller](http://stackoverflow.com/questions/23552296/how-to-determine-hotspot-vm-default-thread-stack-size) than on 64-bit jvm (same for linux). If you are trying to run activator on a 32-bit jvm, make sure to pass -J-Xss1m option to set thread stack size to 1 mb.

**Comment by anikolaev** [comment]
@shkhln, thank you very much. Setting memory option helped. After adding 
`-Xss1M`
 in SBT settings in IDEA they became: 
`-Xmx512M -XX:MaxPermSize=256M -Xss1M`
 and now the project is running successfully. I also added SBT_OPTS right into sbt.bat in case I will need to run it from command line. I was finding somewhere in discussions low memory mentioning in similar problem but I was not sure what to change to use more memory (there was only recommendation to stop some programs which haven't help). 

Sorry for adding issue which is not related to Play project but I had no idea where to look for the reason. Closing it.
